plugins/services/src/js/components/modals/CreateServiceModalForm.tsx

```diff
@@ -98,7 +100,32 @@ const JSONEditor = React.lazy(
     import(/* webpackChunkName: "jsoneditor" */ "#SRC/js/components/JSONEditor")
 );
 
-class CreateServiceModalForm extends React.Component {
+class CreateServiceModalForm extends React.Component<
+  {
+    activeTab?: string;
+    errors: FormError[];
+    expandAdvancedSettings: boolean;
+    handleTabChange: (a: string) => void;
+    inputConfigReducers: unknown;
+    isEdit: boolean;
+    isJSONModeActive: boolean;
+    onChange: (s: ServiceSpec) => void;
+    onConvertToPod: (spec: unknown) => void;
+    onErrorsChange: (e: FormError[]) => void;
+    resetExpandAdvancedSettings: () => void;
+    service: ApplicationSpec;
+    showAllErrors: boolean;
+  },
+  {
+    appConfig: {} | null;
+    baseConfig: unknown;
+    batch: Batch;
+    editedFieldPaths: string[];
+    editingFieldPath: null | string;
+    editorWidth?: number;
+    isPod: boolean;
+  }
```

Comment:
☝️ some typing to make a little more sure we don't break stuff while refactoring.